### PR TITLE
Avoid signaling tabsUpdatedPublisher inside selectTabs and some cleanup

### PR DESCRIPTION
**Pull request by darinwf**
- Avoid signaling `tabsUpdatedPublisher` inside `selectTab`
- Remove most `TabManagerDelegate` methods
- Removes unused restoreAllTabs startup prompt
- Suppress signaling `tabsUpdatedPublisher` from `insertTab` when doing bulk adds
- Fixes bug with `addTabsForURLs` not setting `Tab.tabDelegate`

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -496,64 +489,25 @@ class BrowserViewController: UIViewController, ModalPresenter {
         // config log environment variable
         ClientLogger.shared.env = EnvironmentHelper.shared.env
 
-        if !displayedRestoreTabsAlert && !cleanlyBackgrounded() && crashedLastLaunch() {
```
Comment (evancrow) [inline Client/Frontend/Browser/BrowserViewController.swift:499]: Woo-hoo!

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -496,64 +489,25 @@ class BrowserViewController: UIViewController, ModalPresenter {
...
 
-    fileprivate func showRestoreTabsAlert() {
-        guard tabManager.hasTabsToRestoreAtStartup() else {
-            tabManager.selectTab(tabManager.addTab())
-            return
+        // Handle the case of an existing user upgrading to a version of the app
+        // that supports preview mode. They will have tabs already, so we don't
+        // want to show them the preview home experience.
+        // TODO: This is flawed as an existing user may have closed their tabs.
+        if !tabManager.tabs.isEmpty {
```
Comment (evancrow) [inline Client/Frontend/Browser/BrowserViewController.swift:498]: Maybe we can also look at history/see if app data is empty at all?

File: Client/Frontend/Browser/Tabs/TabManager.swift [hunk 3]
```diff
@@ -16,12 +16,6 @@ protocol TabManagerDelegate: AnyObject {
     func tabManager(
         _ tabManager: TabManager, didSelectedTabChange selected: Tab?, previous: Tab?,
         isRestoring: Bool, updateZeroQuery: Bool)
-    func tabManager(_ tabManager: TabManager, didAddTab tab: Tab, isRestoring: Bool)
```
Comment (evancrow) [inline Client/Frontend/Browser/Tabs/TabManager.swift:19]: !!!

File: ClientTests/TabManagerTests.swift [hunk 4]
```diff
@@ -71,46 +72,31 @@ open class MockTabManagerDelegate: TabManagerDelegate {
...
-    public func tabManagerDidAddTabs(_ tabManager: TabManager) {
-        testDelegateMethodWithName(#function, tabs: [])
-    }
-
-    public func tabManagerDidRemoveAllTabs(_ tabManager: TabManager) {
-        testDelegateMethodWithName(#function, tabs: [])
-    }
 }
 
 class TabManagerTests: XCTestCase {
```
Comment (evancrow) [inline ClientTests/TabManagerTests.swift:78]: Nit: remove extra line

File: ClientTests/TabManagerTests.swift [hunk 5]
```diff
@@ -257,24 +241,25 @@ class TabManagerTests: XCTestCase {
     }
 
     func testDeleteLastTab() {
-
         //create the tab before adding the mock delegate. So we don't have to check delegate calls we dont care about
```
Comment (evancrow) [inline ClientTests/TabManagerTests.swift:244]: nit: space after `//`

File: Shared/SentryIntegration.swift [hunk 6]
```diff
@@ -33,10 +33,6 @@ public enum SentryTag: String {
 public class Sentry {
```
Comment (evancrow) [inline Shared/SentryIntegration.swift:33]: Should probably try to remove sentry, not in this PR though. Unless we still want to use it for some reason

**Review by evancrow — APPROVED** [review]
LGTM, a few nits (some) not necessarily apart of this pr, but can be easily fixed

File: Shared/SentryIntegration.swift (hunk 6 above)
Comment (darinwf) [reply Shared/SentryIntegration.swift:33]: Yes, I think we should nuke all of the Sentry code. If we want to integrate with Sentry in the future, we'll just end up doing it all over again anyways.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (darinwf) [reply Client/Frontend/Browser/BrowserViewController.swift:498]: Yeah, that's a good idea.

**Review by evancrow — APPROVED** [review]

**Pull request merged**